fs/dafsclient/rmtclient.cpp

```diff
@@ -431,29 +435,28 @@ void receiveDaFsBuffer(ISocket * socket, MemoryBuffer & tgt, unsigned numtries,
             size32_t szread;
             SOCKREADTMS(socket)((gotLength<4000)?tgt.reserve(gotLength):tgt.reserveTruncate(gotLength), gotLength, gotLength, szread, timeout);
         }
-        catch (IJSOCK_Exception *e) {
-            if (e->errorCode()!=JSOCKERR_timeout_expired) {
-                EXCLOG(e,"receiveDaFsBuffer(1)");
-                PrintStackReport();
-                if (!tm.timemon||!tm.timemon->timedout())
-                    flushDaFsSocket(socket);
-            }
-            else {
-                EXCLOG(e,"receiveDaFsBuffer");
-                PrintStackReport();
-            }
+        catch (IJSOCK_Exception *e)
+        {
+            exception.setown(e);
             tgt.setLength(origlen);
-            throw;
         }
-        catch (IException *e) {
-            EXCLOG(e,"receiveDaFsBuffer(2)");
-            PrintStackReport();
-            if (!tm.timemon||!tm.timemon->timedout())
-                flushDaFsSocket(socket);
+        catch (IException *e)
+        {
+            exception.setown(e);
             tgt.setLength(origlen);
```

Comment:
prob. worth moving this line (and one on line 441) into the if (exception) block.